### PR TITLE
correct call sites and documentation for millisecond timeout API relative to timeout constants

### PR DESCRIPTION
**Pull request by pabigot**
Some API that takes timeouts represented by integral milliseconds recommended using the K_NO_WAIT and K_FOREVER constants, which are no longer integers.  The script added here replaces the incorrect arguments in call sites with the recommended alternative: 0 for `K_NO_WAIT` and `SYS_FOREVER_MS` for `K_FOREVER`.

Updated to include application of the fix, and manual update of related API documentation.

### CODE REVIEW

**Comment by pabigot** [comment]
The following diagnostics are reported by this script, identifying files that do not build in Zephyr master at the time this was run.

APIs that expect millisecond timeout parameters were identified by looking for `s32_t timeout` in the description.  Some API may be missed.

```
./tests/drivers/uart/uart_async_api/src/test_uart_async.c:406:1-15: WARNING: [msl1] replace constant K_FOREVER with ms duration in uart_rx_enable
./tests/drivers/uart/uart_async_api/src/test_uart_async.c:408:1-8: WARNING: [msl1] replace constant K_FOREVER with ms duration in uart_tx
./tests/drivers/uart/uart_async_api/src/test_uart_async.c:415:1-8: WARNING: [msl1] replace constant K_FOREVER with ms duration in uart_tx
./drivers/can/can_shell.c:201:7-15: WARNING: [msl3] replace constant K_FOREVER with ms duration in can_send
./subsys/canbus/canopen/CO_driver.c:348:7-15: WARNING: [msl3] replace constant K_NO_WAIT with ms duration in can_send
./subsys/canbus/canopen/CO_driver.c:136:9-17: WARNING: [msl3] replace constant K_NO_WAIT with ms duration in can_send
./samples/drivers/lora/receive/src/main.c:50:8-17: WARNING: [msl3] replace constant K_FOREVER with ms duration in lora_recv
./samples/drivers/CAN/src/main.c:260:2-10: WARNING: [msl3] replace constant K_FOREVER with ms duration in can_send
```

File: include/bluetooth/buf.h [hunk 1]
```diff
@@ -57,8 +57,7 @@ enum bt_buf_type {
  *
  *  @param type    Type of buffer. Only BT_BUF_EVT and BT_BUF_ACL_IN are
  *                 allowed.
- *  @param timeout Timeout in milliseconds, or one of the special values
- *                 K_NO_WAIT and K_FOREVER.
+ *  @param timeout Timeout in milliseconds, or 0, or SYS_FOREVER_MS.
```
Comment (joerchan) [inline include/bluetooth/buf.h:60]: This is conflicting with the change here:
https://github.com/zephyrproject-rtos/zephyr/pull/24126/files#diff-3b88339f7f6c1a678793dbd25736263a

I suggest to revert the changes in this file.

**Review by joerchan — CHANGES_REQUESTED** [review]

File: include/bluetooth/buf.h (hunk 1 above)
Comment (pabigot) [reply include/bluetooth/buf.h]: done

File: include/drivers/can.h [hunk 2]
```diff
@@ -372,7 +373,7 @@ static inline int z_impl_can_send(struct device *dev,
  * @param length Number of bytes to write (max. 8).
  * @param id  Identifier of the can message.
  * @param rtr Send remote transmission request or data frame
- * @param timeout Waiting for empty tx mailbox timeout in ms or K_FOREVER
+ * @param timeout Waiting for empty tx mailbox timeout in ms or S_FOREVER_MS
```
Comment (joerchan) [inline include/drivers/can.h:376]: SYS

**Review by joerchan — APPROVED** [review]

**Comment by jhedberg** [comment]
@pabigot are you sure that the places you're changing are not still pending conversion to k_timeout_t before 2.3 is released?

**Review by carlescufi — APPROVED** [review]
LGTM, thanks @pabigot

**Comment by pabigot** [comment]
> @pabigot are you sure that the places you're changing are not still pending conversion to k_timeout_t before 2.3 is released?

@jhedberg AFAIK there's no issue tracking what needs to be fixed, so no, I can't be sure somebody isn't working to change these in a different way.  My goal in this is as it was three weeks ago: to provide the minimum change necessary to make sure there's nothing in tree that is left in an unbuildable state.

All I can do is work on what's in tree, adjusted when somebody tells me there's a conflicting PR.

**Review by jukkar — APPROVED** [review]

**Comment by carlescufi** [comment]
> @pabigot are you sure that the places you're changing are not still pending conversion to k_timeout_t before 2.3 is released?

We will post a PR removing the use of the legacy timeout option soon

**Review by jhedberg — APPROVED** [review]
